### PR TITLE
Add preStop and postStop hooks

### PR DESCRIPTION
**Pull request by justenwalker**
- Refactor hook calling into runCommand method
- Add docs and example for preStop and postStop
- Add runCommand tests

See #28

### CODE REVIEW

**Comment by justenwalker** [comment]
Build fails because of signal race condition - relevant tests do pass though:

```
=== RUN   TestRunCommand
Running doStuff with args: --debug
Running failStuff with args: --debug
2015/12/05 04:08:44 exit status 255
--- PASS: TestRunCommand (0.00s)
```

**Comment by justenwalker** [comment]
Rebased on the #36 fix so the tests would pass and to incorporate the refactoring.

File: src/containerbuddy/config.go [hunk 1]
```diff
@@ -20,8 +20,12 @@ var (
 type Config struct {
 	Consul       string `json:"consul,omitempty"`
 	OnStart      string `json:"onStart"`
+	PreStop      string `json:"preStop"`
+	PostStop     string `json:"postStop"`
 	StopTimeout  int    `json:"stopTimeout"`
 	onStartArgs  []string
+	preStopArgs  []string
+	postStopArgs []string
```
Comment (tgross) [inline src/containerbuddy/config.go]: I did a search thru the code and it looks like `onStartArgs`, `preStopArgs`, and `postStopArgs` are never used. You're passing the equivalent public member of the struct to `runCommand` instead, so there's no reason to have these.

File: src/containerbuddy/main.go [hunk 2]
```diff
@@ -111,6 +112,18 @@ func getCmd(args []string) *exec.Cmd {
 	}
 }
 
+func runCommand(commandString string) (int, error) {
```
Comment (tgross) [inline src/containerbuddy/main.go]: I feel like it makes more sense to move the string splitting into the `run`, and then we can eliminate `ServiceConfig.healthArgs` and `Backend.onChangeArgs` members too (and the splitting code in config.go).

**Comment by tgross** [comment]
cc'ing in @misterbisson here too.

**Comment by justenwalker** [comment]
@tgross incorporated the changes you made in the last update - rebased on latest master.

**Comment by misterbisson** [comment]
lgtm

**Comment by tgross** [comment]
Apparently TravisCI is having some build-time issues. We'll wait for them to catch up and then merge this once we get the +1 from them.

**Comment by justenwalker** [comment]
@tgross - I kicked the build with another rebase and Travis finally tested it

**Comment by tgross** [comment]
Thanks @justenwalker. Merging!

**Pull request merged**